docs/reference/rest-api/logs.asciidoc

```diff
@@ -0,0 +1,278 @@
+[role="xpack"]
+[[logs-api]]
+== Log ingestion API
+
+experimental::[]
+
+Provides a simple JSON API to ingest log events into {es}.
+
+[discrete]
+[[logs-api-request]]
+=== {api-request-title}
+
+`POST /_logs`
+
+`POST /_logs/<dataset>`
+
+`POST /_logs/<dataset>/<namespace>`
+
+[discrete]
+[[logs-api-prereqs]]
+=== {api-prereq-title}
+* If the {es} {security-features} are enabled, you must have the `create`
+<<privileges-list-indices,index privileges>> for the target data stream.
+* As the target stream depends on routing rules and the data provided in the log events,
+it's recommended to grant permissions for `logs-*-*`.
+* Automatic data stream creation requires a matching index template with data
+stream enabled. See <<set-up-a-data-stream>>.
+
+[discrete]
+[[logs-api-desc]]
+=== {api-description-title}
+
+Provides a way to ingest log events into {es}, similar to the <<docs-bulk, Bulk API>>.
+
+The log events are specified in the request body using a newline delimited JSON (NDJSON) structure.
+
+The events are indexed into the `logs-<dataset>-<namespace>` <<data-streams, data stream>>,
+according to the dataset and namespace parameters, which can be provided globally or on a per-event basis.
+
+The endpoint is designed in a way that logs are never dropped, as long as the cluster has enough capacity.
+
+If an error happens during ingestion,
+the logs are sent to the `logs-dlq-<namespace>` data stream that acts as a dead letter queue for failed events.
+However, log ingestion should rarely fail as the mappings in the built-in index template for the `logs-*-*` data streams are designed to minimize mapping conflicts.
+
+A <<logs-api-request-body, couple of fields>> from the {ecs-ref}[Elastic Common Schema (ECS)] are indexed by default that are commonly used to search or filter logs.
+All other fields are not indexed by default.
+But you can still add any top-level fields and use them in searches and aggregations as the default index template for logs
+<<dynamic-mapping-runtime-fields, maps dynamic fields as runtime fields>>.
+
+For custom fields that are frequently used in searches or aggregations, you might want to leverage the speed benefits of indexing the field.
+This comes at the expense of a larger index size more processing at ingest time.
+To index a custom field, <<create-index-template, create an index template>> for your dataset `logs-<dataset>-*` based on the built-in index template for `logs-*-*` and add your custom field to the mappings.
```

Comment:
Ideally that is a part we would have an API for, user should not have to think about templates etc. Same for the rollover below.